sdk/containers/azcontainerregistry/blob_client_test.go

```diff
@@ -8,99 +8,221 @@ package azcontainerregistry
+	require.NotEmpty(t, blobDigest)
+	_, hash, found := strings.Cut(blobDigest, ":")
+	require.True(t, found)
+	if recording.GetRecordMode() == recording.RecordingMode {
+		require.NoError(t, recording.AddGeneralRegexSanitizer("00", hash, nil))
+	}
+	return blobDigest
 }
 
-func TestBlobClient_CancelUpload_fail(t *testing.T) {
-	startRecording(t)
-	endpoint, cred, options := getEndpointCredAndClientOptions(t)
-	ctx := context.Background()
-	client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
-	require.NoError(t, err)
-	_, err = client.CancelUpload(ctx, "wrong location", nil)
-	require.Error(t, err)
+func TestBlobClient(t *testing.T) {
+	repository, digest := buildImage(t)
+	blobDigest := blobDigest(t, repository, digest)
+
+	t.Run("CheckBlobExists", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.CheckBlobExists(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		require.Equal(t, blobDigest, *res.DockerContentDigest)
+	})
+
+	t.Run("CheckChunkExists", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		res, err := client.CheckChunkExists(ctx, repository, blobDigest, "bytes=0-299", nil)
+		require.NoError(t, err)
+		require.NotEmpty(t, *res.ContentLength)
+	})
+
+	t.Run("CheckChunkExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckChunkExists(ctx, repository, "wrong digest", "bytes=0-299", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CheckBlobExists_fail", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		_, err = client.CheckBlobExists(ctx, repository, "wrong digest", nil)
+		require.Error(t, err)
+	})
+
+	t.Run("CompleteUpload_wrongDigest", func(t *testing.T) {
+		startRecording(t)
+		endpoint, cred, options := getEndpointCredAndClientOptions(t)
+		client, err := NewBlobClient(endpoint, cred, &BlobClientOptions{ClientOptions: options})
+		require.NoError(t, err)
+		getRes, err := client.GetBlob(ctx, repository, blobDigest, nil)
+		require.NoError(t, err)
+		blob, err := io.ReadAll(getRes.BlobData)
+		require.NoError(t, err)
+		startRes, err := client.StartUpload(ctx, "hello-world", nil)
+		require.NoError(t, err)
+		uploadResp, err := client.uploadChunk(ctx, *startRes.Location, streaming.NopCloser(bytes.NewReader(blob)), nil)
```

Comment:
Is there no corresponding public function for this? Ie, is this just us sneaking in so we can test something users don't normally look at?

Reply:
Its public face is [UploadChunk](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry#BlobClient.UploadChunk). I didn't write this test, however I would guess the motivation for calling the unexported method was to avoid faking the digest calculator.